iwyu_preprocessor.cc

```diff
@@ -1087,4 +1129,25 @@ bool IwyuPreprocessorInfo::ForwardDeclareIsInhibited(
       ContainsKey(*inhibited_forward_declares, normalized_symbol_name);
 }
 
+bool IwyuPreprocessorInfo::ForwardDeclareInKeepRange(
+    const SourceLocation& loc) const {
+  // Check if the entry exists, shortcircuit if not
+  const FileEntry* file = GetFileEntry(loc);
+
+  auto keep_ranges = keep_location_ranges_.equal_range(file);
+
+  // with multimaps first==second if and only if there's no matches
+  if (keep_ranges.first == keep_ranges.second) {
+    return false;
+  }
+
+  const SourceRange decl_range{loc};
```

Comment:
We typically don't use `const` for local variables, or `{...}` initialization syntax. This would be easier to read as simply `SourceRange decl_range(loc);`